guppy/src/graph/graph_impl.rs

```diff
@@ -1092,137 +1047,235 @@ impl DependencyMetadata {
+        let required = self.required.enabled_on(platform);
+        let optional = self.optional.enabled_on(platform);
+
+        required.or(optional)
+    }
+
+    /// Returns the `PlatformStatus` corresponding to whether this dependency is required.
+    pub fn required_status(&self) -> PlatformStatus<'g> {
+        self.required
+    }
+
+    /// Returns the `PlatformStatus` corresponding to whether this dependency is optional.
+    pub fn optional_status(&self) -> PlatformStatus<'g> {
+        self.optional
+    }
+}
+
+/// The status of a dependency or feature, which is possibly platform-dependent.
+///
+/// This is a sub-status of `EnabledStatus`.
+#[derive(Copy, Clone, Debug)]
+pub enum PlatformStatus<'g> {
+    /// This dependency or feature is never enabled on any platforms.
     Never,
-    /// The status of this dependency is *unknown* because the evaluation involved target features
-    /// whose status is unknown.
+    /// This dependency or feature is always enabled on all platforms.
+    Always,
+    /// The status is platform-dependent.
+    PlatformDependent {
+        /// An evaluator to run queries against.
+        eval: PlatformEval<'g>,
+    },
+}
+
+impl<'g> PlatformStatus<'g> {
+    fn new(specs: &'g PlatformStatusImpl) -> Self {
+        match specs {
+            PlatformStatusImpl::Always => PlatformStatus::Always,
+            PlatformStatusImpl::Specs(specs) => {
+                if specs.is_empty() {
+                    PlatformStatus::Never
+                } else {
+                    PlatformStatus::PlatformDependent {
+                        eval: PlatformEval { specs },
+                    }
+                }
+            }
+        }
+    }
+
+    /// Returns true if this dependency is always enabled on all platforms.
+    pub fn is_always(&self) -> bool {
+        match self {
+            PlatformStatus::Always => true,
+            PlatformStatus::PlatformDependent { .. } | PlatformStatus::Never => false,
+        }
+    }
+
+    /// Returns true if this dependency is never enabled on any platform.
+    pub fn is_never(&self) -> bool {
+        match self {
+            PlatformStatus::Never => true,
+            PlatformStatus::PlatformDependent { .. } | PlatformStatus::Always => false,
+        }
+    }
+
+    /// Evaluates whether this dependency is enabled on the given platform.
+    pub fn enabled_on(&self, platform: &Platform<'g>) -> IsEnabled {
+        match self {
+            PlatformStatus::Never => IsEnabled::Disabled,
+            PlatformStatus::Always => IsEnabled::Enabled,
+            PlatformStatus::PlatformDependent { eval } => eval.eval(platform),
+        }
+    }
+}
+
+/// Whether a dependency or feature is enabled on a specific platform.
+///
+/// Returned by the methods on `EnabledStatus`, `PlatformStatus`, and `PlatformEval`.
+#[derive(Copy, Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub enum IsEnabled {
```

Comment:
Not a fan of this type name, though I have little to offer in constructive suggestions, especially as EnabledStatus is already used.  Maybe a more generic 3v logic type would make more sense like `Ternary` or `Trilean` etc?

Reply:
I'll rename it to `EnabledTernary`